channeldb/invoices.go

```diff
@@ -523,6 +527,9 @@ func validateInvoice(i *Invoice) error {
 	if i.Terms.Features == nil {
 		return errors.New("invoice must have a feature vector")
 	}
+	if len(i.Htlcs) > 0 {
```

Comment:
just something i noticed we weren't strictly enforcing. realized that it you accidentally added an invoice w/ AMP htlcs that it would not be retrievable from the set id index. so this just enforces that htlcs can only be added through `UpdateInvoice`